hexrdgui/masking/mask_manager.py

```diff
@@ -0,0 +1,390 @@
+import math
+import numpy as np
+
+from PySide6.QtCore import Signal, QObject
+from hexrdgui import utils
+
+from hexrdgui.constants import ViewType
+from hexrdgui.masking.constants import MaskType
+from hexrdgui.masking.create_polar_mask import (
+    create_polar_mask_from_raw, rebuild_polar_masks
+)
+from hexrdgui.masking.create_raw_mask import (
+    recompute_raw_threshold_mask, create_raw_mask, rebuild_raw_masks
+)
+from hexrdgui.hexrd_config import HexrdConfig
+from hexrdgui.masking.mask_compatability import load_masks_v1_to_v2
+from hexrdgui.singletons import QSingleton
+from hexrdgui.utils import unique_name
+
+from hexrd.instrument import unwrap_dict_to_h5
+
+from abc import ABC, abstractmethod
+
+
+class Mask(ABC):
+    def __init__(self, name='', mtype='', visible=True):
+        self.type = mtype
+        self.name = name
+        self.visible = visible
+        self.masked_arrays = None
+
+    def update_mask_visibility(self, visibility):
+        self.visible = visibility
+
+    # Abstract methods
+    @abstractmethod
+    def get_data(self):
+        pass
+
+    @abstractmethod
+    def set_data(self, data):
+        pass
```

Comment:
`data` could be a property getter/setter as well, unless you anticipate more than one argument in the future.

Reply:
No, I can't. Updated to be property getter/setter instead